components/AddressShipping/NewAddress.tsx

```diff
@@ -15,53 +15,240 @@ import {COLOR} from '../../src/assets/color';
 import {ARROW_LEFT, ARROW_RIGHT} from '../../src/assets/image';
 import {useNavigation} from '@react-navigation/native';
 import CheckBox from '@react-native-community/checkbox';
+import {HOME_KHO} from '../../src/assets/image';
+import {ButtonGradient} from '../Button/ButtonGradient';
+import {ADDRESS_SCREEN, ADDRESS_SHIPPING} from '../../constant/route';
+import {useSelector} from 'react-redux';
+import {FormService} from '../../common/FormService';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import Geolocation from '@react-native-community/geolocation';
+import {Ship} from '../../services/ShippingService';
+import {useDispatch} from 'react-redux';
+import {clearInfoAddress} from '../../constant/action';
 export const NewAddress: FC = () => {
+  const dispatch = useDispatch();
+  const register = useSelector(store => store.register.register);
   const navigation = useNavigation();
+  const [typeAddress, setTypeAddress] = useState([
+    {
+      id: '1',
+      value: 'kho hàng',
+      isDefault: true,
+    },
+    {
+      id: '2',
+      value: 'Cừa hàng phân phối',
+      isDefault: false,
+    },
+  ]);
+  const [select, setSelect] = useState(true);
+  const [value, setValue] = useState('');
+  const [toggleCheckBox, setToggleCheckBox] = useState(true);
+  const [typeScreen, setTypeScreen] = useState('address_shipping');
+  const [infoContact, setInfoContact] = useState({
+    name_recipient: '',
+    phone: '',
+  });
+  const [errors, setErrors] = useState({
+    name_recipient: '',
+    phone: '',
+  });
+  const [location, setLoccation] = useState({
+    latitude: '',
+    longitude: '',
+  });
+
+  useEffect(() => {
+    getLocation();
+  }, []);
   const goBack = () => {
     navigation.goBack();
   };
+
+  const moveAddressShipping = () => {
+    navigation.navigate(ADDRESS_SCREEN, {
+      type_screen: typeScreen,
+    });
+  };
+
+  const renderOption = (item, index) => {
+    return (
+      <TouchableOpacity
+        style={[
+          styles.button,
+          item.isDefault === select ? styles.checked : styles.unchecked,
+        ]}
+        key={index}
+        onPress={() => selectOption(item)}>
+        <Image style={styles.icon} source={HOME_KHO} />
+        <Text>{item.value}</Text>
+      </TouchableOpacity>
+    );
+  };
+
+  const selectOption = item => {
+    select ? setSelect(false) : setSelect(true);
+    setValue(item.id);
+  };
+  const setCheckBox = () => {
+    toggleCheckBox ? setToggleCheckBox(false) : setToggleCheckBox(true);
```

Comment:
chú ý những chỗ như này code cho bớt nông dân lại
```suggestion
     setToggleCheckBox(! toggleCheckBox)
```